src/mscorlib/shared/System/StringSpanHelpers.cs

```diff
@@ -149,8 +151,38 @@ public static void CheckStringComparison(StringComparison comparisonType)
             // Single comparison to check if comparisonType is within [CurrentCulture .. OrdinalIgnoreCase]
             if ((uint)(comparisonType - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
             {
-                throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
+                ThrowStringComparison_NotSupported();
             }
         }
+
+        public static CompareOptions GetCaseCompareOfComparisonCulture(StringComparison comparisonType)
+        {
+            Debug.Assert((uint)(comparisonType - StringComparison.CurrentCulture) <= (StringComparison.InvariantCultureIgnoreCase - StringComparison.CurrentCulture));
```

Comment:
nit: Is there a way for us to use the CheckStringComparison here to avoid duplication?
Maybe if we change it to return bool and the assert could be CheckStringComparison() == true.